internal/resources/syntheticmonitoring/resource_check.go

```diff
@@ -119,6 +119,23 @@ var (
 				MaxItems:    1,
 				Elem:        syntheticMonitoringCheckSettingsMultiHTTP,
 			},
+			"scripted": {
+				Description: "Settings for scripted check. See https://grafana.com/docs/grafana-cloud/testing/synthetic-monitoring/create-checks/checks/k6/.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				MaxItems:    1,
+				Elem:        syntheticMonitoringCheckSettingsScripted,
+			},
+		},
+	}
+
+	syntheticMonitoringCheckSettingsScripted = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"script": {
+				Type:     schema.TypeString,
+				Optional: false,
+				Required: true,
+			},
```

Comment:
@julienduchesne, Thanks for the quick response but it would be nice if we can take in a file. For example,

instead of 

```
```terraform
data "grafana_synthetic_monitoring_probes" "main" {}
resource "grafana_synthetic_monitoring_check" "scripted" {
  job     = "Scripted defaults"
  target  = "scripted target"
  enabled = false
  probes = [
    data.grafana_synthetic_monitoring_probes.main.probes.Paris,
  ]
  labels = {
    foo = "bar"
  }
  settings {
    scripted {
      script = "console.log('Hello, world!')"
    }
  }
}
```
something like this would be nicer 

```
```terraform
data "grafana_synthetic_monitoring_probes" "main" {}
resource "grafana_synthetic_monitoring_check" "scripted" {
  job     = "Scripted defaults"
  target  = "scripted target"
  enabled = false
  probes = [
    data.grafana_synthetic_monitoring_probes.main.probes.Paris,
  ]
  labels = {
    foo = "bar"
  }
  settings {
    scripted {
      script = "/path/to/js-script.js"
    }
  }
}
```
Happy to look into this and contribute a Pr if needed.